x/poolmanager/keeper.go

```diff
@@ -45,9 +47,13 @@ type Keeper struct {
 
 	defaultTakerFeeBz  []byte
 	defaultTakerFeeVal osmomath.Dec
+
+	cachedTakerFeeShareAgreementMap          map[string]types.TakerFeeShareAgreement
+	cachedRegisteredAlloyPoolByAlloyDenomMap map[string]types.AlloyContractTakerFeeShareState
+	cachedRegisteredAlloyedPoolIdArray       []uint64
```

Comment:
I think that we shouls document the behavior of the cache in the README and make a link/reference to the documention here

Reply:
cache documentation added here https://github.com/osmosis-labs/osmosis/pull/8310/commits/bb2903cffa2f50d9adbc79c251f248be455c3e4a